DirectProgramming/DPC++FPGA/include/unrolled_loop.hpp

```diff
@@ -0,0 +1,187 @@
+#ifndef __UNROLLEDLOOP_HPP__
+#define __UNROLLEDLOOP_HPP__
+
+#include <type_traits>
+#include <utility>
+
+namespace fpga_tools {
+
+//
+// The code below creates the constexprs 'make_integer_range'
+// and 'make_index_range' these are akin to 'std::make_integer_sequence'
+// and 'std::make_index_sequence', respectively.
+// However they allow you to specificy a range and can either increment
+// or decrement, rather than a strict increasing sequence
+//
+template <typename T, typename, T begin, bool increase>
+struct integer_range_impl;
```

Comment:
Naming convention for functions/member functions not consistent with rest of source code. Is this different due to emulating a C++ standard library?

Reply:
As mentioned in the comment, these classes are meant to mirror std::make_integer_sequence and std::make_integer_range, which is why they do not follow our normal naming conventions.